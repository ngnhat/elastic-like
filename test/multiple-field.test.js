const Store = require('../index');

describe('single field', () => {
  it('multiple documents', () => {
    const store = new Store();

    store.add({ Id: 1, Code: 'aaa', Name: 'aaa bbb ccc' });
    store.add({ Id: 2, Code: 'bbb', Name: 'bbb ccc ddd' });
    store.add({ Id: 3, Code: 'ccc', Name: 'ccc ddd eee' });
    store.add({ Id: 4, Code: 'ddd', Name: 'ddd eee fff' });

    expect(store.search('ccc')).toMatchSnapshot();
  });

  it('multiple documents - multiple terms', () => {
    const store = new Store();
    store.add({ Id: 1, Code: 'aaa', Name: 'aaa bbb ccc' });
    store.add({ Id: 2, Code: 'bbb', Name: 'bbb ccc ddd' });
    store.add({ Id: 3, Code: 'ccc', Name: 'ccc ddd eee' });
    store.add({ Id: 4, Code: 'ddd', Name: 'ddd eee fff' });

    expect(store.search('bbb aaa')).toMatchSnapshot();
    expect(store.search('ccc fff')).toMatchSnapshot();
  });
});
