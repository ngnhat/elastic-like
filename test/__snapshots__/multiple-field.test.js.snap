// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single field multiple documents - multiple terms 1`] = `
Array [
  Object {
    "score": 1.8971199848858813,
    "source": Object {
      "Code": "aaa",
      "Id": 1,
      "Name": "aaa bbb ccc",
    },
  },
]
`;

exports[`single field multiple documents - multiple terms 2`] = `Array []`;

exports[`single field multiple documents 1`] = `
Array [
  Object {
    "score": 1.5606477482646686,
    "source": Object {
      "Code": "ccc",
      "Id": 3,
      "Name": "ccc ddd eee",
    },
  },
  Object {
    "score": 0.3566749439387324,
    "source": Object {
      "Code": "aaa",
      "Id": 1,
      "Name": "aaa bbb ccc",
    },
  },
  Object {
    "score": 0.3566749439387324,
    "source": Object {
      "Code": "bbb",
      "Id": 2,
      "Name": "bbb ccc ddd",
    },
  },
]
`;
